### PR TITLE
8299191: Unnecessarily global friend functions for relocInfo

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this small cleanup around the relocInfo class.  It declares a
couple of global functions as friends, so they have access to private
constructors and helper functions.  But there is no reason for these functions
to be at global scope.  It is more natural for them to be static factory
functions in relocInfo.

Testing:
mach5 tier1

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299191](https://bugs.openjdk.org/browse/JDK-8299191): Unnecessarily global friend functions for relocInfo

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [90893679](https://git.openjdk.org/jdk/pull/11753/files/90893679ef5274e37deebefb10ff21606cc67fbf)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [90893679](https://git.openjdk.org/jdk/pull/11753/files/90893679ef5274e37deebefb10ff21606cc67fbf)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11753/head:pull/11753` \
`$ git checkout pull/11753`

Update a local copy of the PR: \
`$ git checkout pull/11753` \
`$ git pull https://git.openjdk.org/jdk pull/11753/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11753`

View PR using the GUI difftool: \
`$ git pr show -t 11753`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11753.diff">https://git.openjdk.org/jdk/pull/11753.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/code/relocInfo.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ relocInfo* relocInfo::finish_prefix(short* prefix_limit) {
     return this+1;
   }
   // cannot compact, so just update the count and return the limit pointer
-  (*this) = prefix_relocInfo(plen);   // write new datalen
+  (*this) = prefix_info(plen);       // write new datalen
```
Comment (chhagedorn) [inline src/hotspot/share/code/relocInfo.cpp:89]: Just a minor thing: Any specific reason for these additional whitespaces?

**Review by chhagedorn — APPROVED** [review]
Looks good!

**Review by vnkozlov — APPROVED** [review]
Good.

File: src/hotspot/share/code/relocInfo.cpp (hunk 1 above)
Comment (kimbarrett) [reply src/hotspot/share/code/relocInfo.cpp:89]: No, I've no idea where those spaces came from.  I thought maybe I'd hit `meta-;` (emacs indent for comment), but nope, that puts a different number of spaces there.  And it's not manually lined up with other comments in the function.

Comment (chhagedorn) [reply src/hotspot/share/code/relocInfo.cpp:89]: Okay :-)

**Comment by kimbarrett** [comment]
Thanks for reviews @vnkozlov and @chhagedorn

**Pull request closed without merging**